tests/trsa

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash -e
+# Copyright (C) 2022 Simo Sorce <simo@redhat.com>
+# SPDX-License-Identifier: Apache-2.0
+
+source "${TESTSSRCDIR}/helpers.sh"
+
+title PARA "Raw Sign"
+#Ensure the first byte is small otherwise there is a 50/50 change the random
+#value will exceed the numeric value of the modulus.
+dd if=/dev/zero of="${TMPPDIR}/255Brandom.bin" bs=1 count=1 >/dev/null 2>&1
+dd if=/dev/urandom of="${TMPPDIR}/255Brandom.bin" seek=1 bs=1 count=255 >/dev/null 2>&1
+ossl '
+rsautl -sign -inkey "${PRIURI}" -raw
+              -in ${TMPPDIR}/255Brandom.bin
+              -out ${TMPPDIR}/raw-sig.bin'
+
+title PARA "Raw Verify"
+ossl '
+rsautl -verify -inkey "${PUBURI}" -pubin -raw
+              -in ${TMPPDIR}/raw-sig.bin
+              -out ${TMPPDIR}/raw-text.out'
+diff "${TMPPDIR}/255Brandom.bin" "${TMPPDIR}/raw-text.out"
+
+
+title PARA "Raw Sign check error"
+dd if=/dev/urandom of="${TMPPDIR}/64Brandom.bin" bs=64 count=1 >/dev/null 2>&1
+FAIL=0
+ossl '
+pkeyutl -sign -inkey "${BASEURI}"
+              -pkeyopt pad-mode:none
+              -in ${TMPPDIR}/64Brandom.bin
+              -out ${TMPPDIR}/no-raw-sig.bin' || FAIL=1
+if [ $FAIL -eq 0 ]; then
+    echo "Raw signature should not allow data != modulus size"
+    exit 1
+fi
+# unfortunately pkeyutl simply does not make it possible to sign anything
+# that is bigger than a hash, which means we'd need a very small RSA key
+# to really check a raw signature through pkeyutl
```

Comment:
is this comment still relevant? I think this was addressed by the use of rsautl above.

Reply:
It is, rsautl is deprecated and will eventually go away, when that happens we'll have to write our own util to test raw keys as openssl pkeyutl just doesn't work